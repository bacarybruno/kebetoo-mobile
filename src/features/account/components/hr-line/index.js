import React from 'react'
import { View } from 'react-native'

import Typography, { types } from '@app/shared/components/typography'

import styles from './styles'

const HrLine = ({ text, style }) => (
  <View style={[styles.wrapper, style]}>
    <View style={styles.line} />
    {text && <Typography type={types.separator} text={text} style={styles.text} />}
    <View style={styles.line} />
  </View>
)

export default HrLine
