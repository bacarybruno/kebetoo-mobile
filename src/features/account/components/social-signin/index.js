import React, { useCallback } from 'react'
import { View, TouchableOpacity, Image } from 'react-native'

import HrLine from '@app/features/account/components/hr-line'
import { googleLogin, facebookLogin } from '@app/shared/helpers'
import Kebeticon from '@app/shared/icons/kebeticons'
import { colors, images } from '@app/theme'
import { useAnalytics } from '@app/shared/hooks'

import styles from './styles'

const SocialSignIn = ({
  sectionText, children, onSignIn, onLoading, disabled, type,
}) => {
  const { trackSignIn, trackSignUp, reportError } = useAnalytics()

  const trackAuthEvent = useCallback((provider) => {
    if (type === 'signIn') {
      trackSignIn(provider)
    } else {
      trackSignUp(provider)
    }
  }, [trackSignIn, trackSignUp, type])

  const signInWithGoogle = useCallback(async () => {
    if (disabled) return false
    onLoading(true)
    const result = await googleLogin()
    if (result.error) {
      reportError(result.error)
    } else {
      trackAuthEvent('google.com')
    }
    onLoading(false)
    return onSignIn(result)
  }, [onLoading, onSignIn, disabled, trackAuthEvent, reportError])

  const signInWithFacebook = useCallback(async () => {
    if (disabled) return false
    onLoading(true)
    const result = await facebookLogin()
    if (result.error) {
      reportError(result.error)
    } else {
      trackAuthEvent('facebook.com')
    }
    onLoading(false)
    return onSignIn(result)
  }, [onLoading, onSignIn, disabled, trackAuthEvent, reportError])

  return (
    <View style={styles.socialSignUp}>
      <View style={styles.socialSignUpContainer}>
        <View style={styles.socialSignUpContent}>
          <HrLine text={sectionText} />
          <View style={styles.socialSignUpButtons}>
            <TouchableOpacity testID="facebook-signin" onPress={signInWithFacebook}>
              <Kebeticon name="facebook" color={colors.facebook} size={33} />
            </TouchableOpacity>
            <TouchableOpacity testID="google-signin" onPress={signInWithGoogle}>
              <Image style={styles.googleIcon} source={images.google_icon} />
            </TouchableOpacity>
          </View>
        </View>
        {children}
      </View>
    </View>
  )
}

SocialSignIn.defaultProps = {
  onSignIn: () => {},
  onLoading: () => {},
}

export default SocialSignIn
