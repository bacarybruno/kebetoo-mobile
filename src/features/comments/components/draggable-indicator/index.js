import { memo } from 'react';
import { View } from 'react-native';

import { useAppStyles } from '@app/shared/hooks';

import createThemedStyles from './styles';

const DraggableIndicator = () => {
  const styles = useAppStyles(createThemedStyles);
  return (
    <View style={styles.draggableContainer}>
      <View style={styles.draggableIcon} />
    </View>
  );
};

export default memo(DraggableIndicator);
