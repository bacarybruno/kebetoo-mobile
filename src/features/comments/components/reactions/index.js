import React, { useEffect, useState, useMemo } from 'react'
import { View } from 'react-native'

import BaseReactions from '@app/features/post/containers/reactions'
import Typography, { types } from '@app/shared/components/typography'
import { usePosts } from '@app/shared/hooks'
import strings from '@app/config/strings'
import Avatar from '@app/shared/components/avatar'

import DraggableIndicator from '../draggable-indicator'
import styles, { reactionsHeight, summaryHeight } from './styles'

export const SummaryAuthor = ({ author }) => {
  if (!author) return <View style={styles.img} />
  return (
    <Avatar src={author.photoURL} text={author.displayName} style={styles.img} />
  )
}

const Summary = React.memo(({ comments }) => {
  const [authors, setAuthors] = useState({})
  const [reactionsMap, setReactionsMap] = useState({})

  const { getAuthors } = usePosts()

  useEffect(() => {
    const reactionsMapData = {}
    comments.flatMap((comment) => comment.reactions).forEach((reaction) => {
      const currentCount = reactionsMapData[reaction.author]
      if (currentCount === undefined) {
        reactionsMapData[reaction.author] = 0
      } else {
        reactionsMapData[reaction.author] = currentCount + 1
      }
    })
    setReactionsMap(reactionsMapData)
  }, [comments])

  useEffect(() => {
    const fetchAuthors = async () => {
      const authorsToFetch = Object.keys(reactionsMap)
      const data = await getAuthors(authorsToFetch)
      setAuthors(data)
    }
    fetchAuthors()
  }, [getAuthors, reactionsMap])

  const reactors = Object.keys(reactionsMap)
  if (reactors.length === 0) return null

  // eslint-disable-next-line no-return-assign
  const sortedReactors = reactors.sort((a, b) => reactionsMap[b] - reactionsMap[a])

  return (
    <View style={styles.summary}>
      <View style={styles.imgs}>
        {sortedReactors[0] && (
          <View style={styles.imgWrapper}>
            <SummaryAuthor author={authors[sortedReactors[0]]} />
          </View>
        )}
        {sortedReactors[1] && (
          <View style={[styles.imgWrapper, styles.img2Wrapper]}>
            <SummaryAuthor author={authors[sortedReactors[1]]} />
          </View>
        )}
      </View>
      <Typography
        type={types.headline5}
        text={strings.formatString(
          strings.comments.people_reacted, reactors.length,
        )}
      />
    </View>
  )
})

const Reactions = ({
  post, author, comments, ...reactionProps
}) => {
  const hasReactions = comments.flatMap((comment) => comment.reactions).length > 0
  const getContainerStyle = useMemo(() => () => {
    let height = reactionsHeight
    if (hasReactions) height += summaryHeight
    return { ...styles.reactionsContainer, height }
  }, [hasReactions])

  return (
    <View style={getContainerStyle()}>
      <DraggableIndicator />
      <View style={styles.reactions}>
        <BaseReactions post={post} author={author} comments={comments} {...reactionProps} />
        {hasReactions && <Summary comments={comments} />}
      </View>
    </View>
  )
}

export default React.memo(Reactions)
