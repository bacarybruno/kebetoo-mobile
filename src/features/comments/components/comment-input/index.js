import React, {
  useState, useCallback, useRef, useEffect,
} from 'react'
import { Animated, View } from 'react-native'

import { EmojiTextInput, AudioPlayer } from '@app/shared/components'
import { strings } from '@app/config'
import { readableSeconds } from '@app/shared/helpers/dates'
import { useAppStyles } from '@app/shared/hooks'

import { SendButton, RecordButton } from '../send-button'
import ReplyInfo from '../reply-info'
import createThemedStyles from './styles'

const baseReplyInfoSize = 62

export const CommentInput = ({
  onChange, onSend, inputRef, value, audioRecorder, isLoading, reply, onReplyClose, ...inputProps
}) => {
  const styles = useAppStyles(createThemedStyles)

  const [inputHeight, setInputHeight] = useState(styles.textInputSize.minHeight)
  const inputSize = useRef(new Animated.Value(0))

  useEffect(() => {
    Animated.timing(inputSize.current, {
      toValue: inputHeight + (reply ? baseReplyInfoSize : 0),
      useNativeDriver: false,
      duration: 200,
    }).start()
  }, [inputHeight, reply])

  const updateInputHeight = useCallback((event) => {
    setInputHeight(
      Math.max(
        styles.textInputSize.minHeight,
        event.nativeEvent.contentSize.height,
      ),
    )
  }, [])

  // FIXME: handle ios keyboard on comments
  return (
    <View style={styles.commentInputWrapper}>
      <View style={styles.flexible}>
        {!audioRecorder.hasRecording && (
          <EmojiTextInput
            multiline
            fieldName="comment"
            placeholder={
              audioRecorder.isRecording
                ? `${strings.comments.recording} (${readableSeconds(audioRecorder.elapsedTime)})`
                : strings.comments.add_comment
            }
            onValueChange={onChange}
            value={value}
            ref={inputRef}
            textStyle={styles.textInputSize}
            editable={!isLoading}
            onContentSizeChange={updateInputHeight}
            height={inputSize.current}
            wrapperStyle={[
              styles.textInputSize,
              styles.textInputWrapper,
              reply && styles.textInputWrapperWithReply,
            ]}
            {...inputProps}
          >
            {reply && (
              <ReplyInfo info={reply} size={baseReplyInfoSize} onClose={onReplyClose} />
            )}
          </EmojiTextInput>
        )}
        {audioRecorder.hasRecording && (
          <View style={styles.audioWrapper}>
            <AudioPlayer
              source={audioRecorder.recordUri}
              onDelete={audioRecorder.reset}
              style={styles.audioPlayer}
              duration={audioRecorder.elapsedTime}
              round
            />
          </View>
        )}
      </View>
      {value?.length === 0 && !audioRecorder.hasRecording
        ? (
          <View>
            <RecordButton
              isRecording={audioRecorder.isRecording}
              start={audioRecorder.start}
              stop={audioRecorder.stop}
            />
          </View>
        ) : (
          <SendButton onPress={onSend} isLoading={isLoading} testID="send-button" />
        )}
    </View>
  )
}

export default React.memo(CommentInput)
