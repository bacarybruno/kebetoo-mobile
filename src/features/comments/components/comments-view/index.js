import {
  memo, useCallback, useMemo, useContext, useEffect,
} from 'react'
import { View, KeyboardAvoidingView, Platform } from 'react-native'
import { FlatList } from 'react-native-gesture-handler'
import { getBottomSpace } from 'react-native-iphone-x-helper'
import { useSafeAreaInsets } from 'react-native-safe-area-context'

import { NoContent } from '@app/shared/components'
import { strings } from '@app/config'
import { useAppColors, useAppStyles, useUser } from '@app/shared/hooks'
import { metrics } from '@app/theme'
import { SafeAreaContext } from '@app/shared/contexts'
import CommentInput from '@app/features/comments/components/comment-input'
import Comment from '@app/features/comments/components/comment'
import SwipeableComment from '@app/features/comments/components/swipeable'

import createThemedStyles from './styles'

export const NoComments = () => (
  <NoContent title={strings.general.no_content} text={strings.comments.no_content} />
)

export const CommentReply = ({ reply, profile, navigation }) => {
  const styles = useAppStyles(createThemedStyles)
  return (
    <View style={styles.replyWrapper}>
      <Comment
        item={reply}
        user={profile.uid}
        authorId={reply.author.id}
        displayName={reply.author.displayName}
        photoURL={reply.author.photoURL}
        avatarSize={30}
        navigation={navigation}
      />
    </View>
  )
}

const CommentItem = ({
  item, replies, toReply, onSetReply, loadReplies, navigation,
}) => {
  const styles = useAppStyles(createThemedStyles)

  const { profile } = useUser()

  const checkSelectedComment = (comment) => (
    toReply?.id === comment.id ? styles.selectedComment : {}
  )

  const onShowReplies = () => loadReplies(item)

  if (!item.author) return null

  return (
    <View style={styles.comment}>
      <SwipeableComment
        style={[styles.swipeable, checkSelectedComment(item)]}
        onFulfilled={() => onSetReply(item)}
      >
        <Comment
          item={item}
          user={profile.uid}
          authorId={item.author.id}
          displayName={item.author.displayName}
          photoURL={item.author.photoURL}
          repliesCount={item.replies?.length}
          onShowReplies={onShowReplies}
          navigation={navigation}
        />
      </SwipeableComment>
      {replies[item.id]?.map((reply) => (
        <SwipeableComment
          onFulfilled={() => onSetReply(reply)}
          style={[styles.swipeableReply, checkSelectedComment(reply)]}
          key={`comment-reply-${reply.id}`}
        >
          <CommentReply reply={reply} profile={profile} navigation={navigation} />
        </SwipeableComment>
      ))}
    </View>
  )
}

// TODO: paginate comments
const CommentsView = ({
  navigation,
  scrollView,
  renderHeader,
  commentInput,
  audioRecorder,

  // hook items
  clearToReply,
  loadReplies,
  onSend,
  onSetReply,
  setComment,
  comments,
  isLoading,
  toReply,
  replies,
  comment,
}) => {
  const styles = useAppStyles(createThemedStyles)
  const { colors } = useAppColors()

  const { updateTopSafeAreaColor, resetStatusBars } = useContext(SafeAreaContext)
  const insets = useSafeAreaInsets()

  useEffect(() => {
    updateTopSafeAreaColor(colors.backgroundSecondary)
    return resetStatusBars
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const renderComment = useMemo(() => ({ item }) => (
    <CommentItem
      item={item}
      replies={replies}
      toReply={toReply}
      onSetReply={onSetReply}
      loadReplies={loadReplies}
      navigation={navigation}
    />
  ), [replies, toReply, loadReplies, onSetReply, navigation])

  const keyExtractor = useCallback((item, index) => `comment-${item?.id}-${index}`, [])

  return (
    <View style={styles.wrapper}>
      <View style={styles.flexible}>
        <FlatList
          data={comments}
          renderItem={renderComment}
          keyExtractor={keyExtractor}
          ListHeaderComponent={renderHeader}
          ListEmptyComponent={NoComments}
          contentContainerStyle={styles.flatlistContent}
          removeClippedSubviews
          ref={scrollView}
        />
      </View>
      <KeyboardAvoidingView
        style={styles.keyboard}
        enabled={Platform.OS === 'ios'}
        keyboardVerticalOffset={insets.bottom + metrics.marginVertical + getBottomSpace()}
        behavior="padding"
      >
        <CommentInput
          onChange={setComment}
          onSend={onSend}
          inputRef={commentInput}
          audioRecorder={audioRecorder}
          value={comment}
          isLoading={isLoading}
          reply={toReply}
          onReplyClose={clearToReply}
        />
      </KeyboardAvoidingView>
    </View>
  )
}

export default memo(CommentsView)
