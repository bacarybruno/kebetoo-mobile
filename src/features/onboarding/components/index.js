import React from 'react'
import { View } from 'react-native'

import { FormatedTypography, Typography } from '@app/shared/components'
import { metrics } from '@app/theme'

import styles from './styles'

const OnboardingSlide = ({ slideTitle, slideDescription }) => {
  const showTitle = metrics.screenHeight > 568
  return (
    <View style={styles.wrapper}>
      {showTitle && (
        <Typography
          bold
          text={slideTitle}
          type={Typography.types.headline1}
          color="black"
          numberOfLines={2}
          adjustsFontSizeToFit
        />
      )}
      <View style={styles.description}>
        <FormatedTypography
          text={slideDescription}
          type={Typography.types.subheading}
          color="black"
        />
      </View>
    </View>
  )
}

export default React.memo(OnboardingSlide)
