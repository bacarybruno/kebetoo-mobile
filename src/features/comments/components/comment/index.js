import { useCallback, useState, useEffect } from 'react';
import { View } from 'react-native';
import Ionicon from 'react-native-vector-icons/Ionicons';
import dayjs from 'dayjs';
import { useNavigation } from '@react-navigation/native';
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';

import {
  CommentPlaceholder, Avatar, Typography, AudioPlayer, Pressable, MultipleTapHandler,
} from '@app/shared/components';
import { getPostType, POST_TYPES } from '@app/features/post/containers/basic-post/helpers';
import { REACTION_TYPES } from '@app/features/post/containers/reactions/helpers';
import { api } from '@app/shared/services';
import { extractMetadataFromName } from '@app/shared/hooks/audio-recorder';
import routes from '@app/navigation/routes';
import { env } from '@app/config';
import Kebeticon from '@app/shared/icons/kebeticons';
import { useAppColors, useAppStyles } from '@app/shared/hooks';
import { readableNumber } from '@app/shared/helpers/strings';

import createThemedStyles from './styles';

export const getAudioSource = (url) => (
  url.startsWith('http')
    ? url
    : `${env.assetsBaseUrl}/${url.startsWith('/') ? url.substr(1) : url}`
);

export const Reactions = ({
  onReaction, reactions, user, repliesCount, onShowReplies,
}) => {
  const loved = reactions.find((reaction) => (
    reaction.author === user && reaction.type === REACTION_TYPES.LOVE
  ));

  const styles = useAppStyles(createThemedStyles);
  const { colors } = useAppColors();

  return (
    <View style={styles.reactionsWrapper}>
      {repliesCount > 0 && (
        <Pressable borderless style={styles.reactionsButton} onPress={onShowReplies}>
          <Typography
            type={Typography.types.headline5}
            text={readableNumber(repliesCount)}
            systemColor={Typography.colors.primary}
          />
          <Typography type={Typography.types.body} text=" " />
          <Kebeticon color={colors.textPrimary} size={14} name="comment" />
        </Pressable>
      )}
      <Pressable
        borderless
        foreground
        style={styles.reactionsButton}
        onPress={() => onReaction(REACTION_TYPES.LOVE)}
        testID="reaction-button"
      >
        <Typography
          type={Typography.types.headline5}
          text={reactions.length > 0 ? reactions.length : null}
          systemColor={Typography.colors.primary}
        />
        <Typography type={Typography.types.body} text=" " />
        {!loved
          ? <Kebeticon testID="reaction" color={colors.textPrimary} size={14} name="heart" />
          : (
            <Ionicon
              name="md-heart"
              testID="reaction"
              color={colors.heart}
              size={Typography.fontSizes.md}
            />
          )}
      </Pressable>
    </View>
  );
};

const Header = ({ displayName, updatedAt }) => {
  const styles = useAppStyles(createThemedStyles);

  return (
    <View style={styles.header}>
      <Typography type={Typography.types.headline5} text={displayName} />
      <Typography type={Typography.types.headline5} text=" â€¢ " />
      <Typography type={Typography.types.headline6} text={dayjs(updatedAt).fromNow()} />
    </View>
  );
};

const Content = ({ item }) => {
  const styles = useAppStyles(createThemedStyles);

  switch (getPostType(item)) {
    case POST_TYPES.AUDIO:
      return (
        <AudioPlayer
          style={styles.audio}
          source={getAudioSource(item.audio.url)}
          duration={parseInt(extractMetadataFromName(item.audio.name).duration, 10)}
        />
      );
    case POST_TYPES.TEXT:
      return <Typography type={Typography.types.body} text={item.content.trim()} />;
    default:
      return null;
  }
};

const Comment = ({
  item,
  displayName,
  photoURL,
  user,
  authorId,
  repliesCount,
  onShowReplies,
  avatarSize = 35,
  navigation,
}) => {
  const [reactions, setReactions] = useState(item.reactions);

  useEffect(() => {
    setReactions(item.reactions);
  }, [item.reactions]);

  // eslint-disable-next-line react-hooks/rules-of-hooks
  const { navigate } = navigation || useNavigation();

  const styles = useAppStyles(createThemedStyles);

  // TODO: optimistic ui update
  const onReaction = useCallback(async (type) => {
    const userReaction = reactions.find((r) => r.author === user);
    if (userReaction === undefined) {
      const result = await api.reactions.createCommentReaction(type, item.id, user);
      result.author = result.author.id;
      setReactions((values) => values.concat([result]));
    } else if (userReaction.type === type) {
      await api.reactions.delete(userReaction.id);
      setReactions((values) => values.filter((r) => r.id !== userReaction.id));
    }
    // // Will be used when we'll have many reactions for comments
    // else {
    //   await api.reactions.edit(userReaction.id, type)
    //   setReactions((values) => {
    //     const reaction = values.find((r) => r.id === userReaction.id)
    //     reaction.type = type
    //     values.map((v) => (v.id === item.id ? reaction : v))
    //     return [...values]
    //   })
    // }
  }, [item.id, reactions, user]);

  const onDoublePress = useCallback(async () => onReaction(REACTION_TYPES.LOVE), [onReaction]);

  const onPress = useCallback(async () => {
    if (onShowReplies) {
      await onShowReplies();
    }
  }, [onShowReplies]);

  const onShowProfile = useCallback(() => {
    navigate(routes.USER_PROFILE, { userId: authorId });
  }, [navigate, authorId]);

  if (!displayName?.trim().length > 0) return <CommentPlaceholder />;

  return (
    <View style={styles.row}>
      <TouchableWithoutFeedback onPress={onShowProfile}>
        <View style={styles.avatarWrapper}>
          <Avatar src={photoURL} text={displayName} size={avatarSize} />
        </View>
      </TouchableWithoutFeedback>
      <MultipleTapHandler onPress={onPress} onDoublePress={onDoublePress}>
        <View style={styles.flexible}>
          <Header displayName={displayName} updatedAt={item.updatedAt} />
          <Content item={item} />
          <Reactions
            reactions={reactions}
            user={user}
            onReaction={onReaction}
            repliesCount={repliesCount}
            onShowReplies={onShowReplies}
          />
        </View>
      </MultipleTapHandler>
    </View>
  );
};

export default Comment;
