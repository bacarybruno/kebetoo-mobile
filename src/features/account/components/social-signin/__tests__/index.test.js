import React from 'react'
import { fireEvent } from 'react-native-testing-library'
import { act } from 'react-test-renderer'
import auth from '@react-native-firebase/auth'

import setupTest from '@app/config/jest-setup'
import { strings } from '@app/config'
import { Typography } from '@app/shared/components'

import SocialSignin, { SocialSignInSection } from '../index'

const givenSocialSignin = setupTest(SocialSignin)({
  text: strings.auth.or_signin_with,
  onSignIn: jest.fn(),
  children: <Typography text={strings.auth.dont_have_account} type={Typography.types.textButton} />,
  type: 'signIn',
  enabled: true,
})

it('renders enabled SocialSignin', () => {
  const { wrapper } = givenSocialSignin()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders disabled SocialSignIn', () => {
  const { wrapper } = givenSocialSignin({ enabled: false })
  expect(wrapper.root.findAllByType(SocialSignInSection).length).toBe(0)
})

describe('sign in', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const onSignInPayload = expect.objectContaining({
    data: auth().credentials,
    error: null,
  })

  const testProvider = async (testID, onSignIn, type) => {
    const { wrapper, props } = givenSocialSignin({
      onSignIn,
      type,
    })
    const signinButton = wrapper.root.findByProps({ testID })
    await act(async () => {
      await fireEvent.press(signinButton)
    })
    return props
  }

  test('with google', async () => {
    const onSignIn = jest.fn()
    await testProvider('google-signin', onSignIn, 'signIn')
    expect(onSignIn).toBeCalledTimes(1)
    expect(onSignIn).toBeCalledWith(onSignInPayload)
  })

  test('with facebook', async () => {
    const onSignIn = jest.fn()
    await testProvider('facebook-signin', onSignIn, 'signUp')
    expect(onSignIn).toBeCalledTimes(1)
    expect(onSignIn).toBeCalledWith(onSignInPayload)
  })

  test('without handler', async () => {
    const props = await testProvider('facebook-signin', undefined)
    expect(props.onSignIn).not.toBeDefined()
  })
})
