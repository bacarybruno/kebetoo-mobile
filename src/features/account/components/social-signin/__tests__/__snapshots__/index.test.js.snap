// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders enabled SocialSignin 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(229, 229, 234, 255)",
              "borderWidth": 0.5,
              "flex": 1,
              "height": 0.5,
            }
          }
        />
        <View
          style={
            Object {
              "borderColor": "rgba(229, 229, 234, 255)",
              "borderWidth": 0.5,
              "flex": 1,
              "height": 0.5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 255)",
              "borderRadius": 10,
              "opacity": 1,
            }
          }
          testID="facebook-signin"
        >
          <Text
            allowFontScaling={false}
            style={
              Object {
                "color": "#3b5998",
                "fontFamily": "icomoon",
                "fontSize": 33,
                "fontStyle": "normal",
                "fontWeight": "normal",
              }
            }
            text="kebeticon-facebook"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="google-signin"
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/google.png",
              }
            }
            style={
              Object {
                "height": 33,
                "marginHorizontal": 8,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": -0.32,
            "lineHeight": 21,
          },
          Array [
            Object {
              "color": "rgba(0,0,0,0.87)",
            },
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            undefined,
          ],
        ]
      }
    >
      Don't have an account?
    </Text>
  </View>
</View>
`;
